docs/parallel.md

```diff
@@ -32,166 +32,65 @@ By default the tests are assigned one by one to the avaible workers this may lea
+Create a placeholder in file:
 
-[basic:firefox] -- FAILURES:
+```js
+#!/usr/bin/env node
+const { Workers } = require('codeceptjs');
+// here will go magic
 ```
 
-### Hooks
-
-Hooks are available when using the `run-multiple` command to perform actions before the test suites start and after the test suites have finished. See [Hooks](/hooks/#bootstrap-teardown) for an example.
-
-
-### Parallel Execution
-
-CodeceptJS can be configured to run tests in parallel.
-
-When enabled, it collects all test files and executes them in parallel by the specified amount of chunks. Given we have five test scenarios (`a_test.js`,`b_test.js`,`c_test.js`,`d_test.js` and `e_test.js`), by setting `"chunks": 2` we tell the runner to run two suites in parallel. The first suite will run `a_test.js`,`b_test.js` and `c_test.js`, the second suite will run `d_test.js` and `e_test.js`.
+Now let's see how to update this file for different parallelization modes:
 
+### Example: Running tests in 2 browsers in 4 threads
 
 ```js
-multiple: {
-  parallel: {
-    // Splits tests into 2 chunks
-    chunks: 2
-  }
-}
-```
+const workerConfig = {
+  testConfig: './test/data/sandbox/codecept.customworker.js',
+};
 
-To execute them use `run-multiple` command passing configured suite, which is `parallel` in this example:
+// don't initialize workers in constructor
+const workers = new Workers(null, workerConfig);
+// split tests by suites in 2 groups
+const testGroups = workers.createGroupsOfSuites(2);
 
-```
-codeceptjs run-multiple parallel
-```
+const browsers = ['firefox', 'chrome'];
 
-Grep and multiple browsers are supported. Passing more than one browser will multiply the amount of suites by the amount of browsers passed. The following example will lead to four parallel runs.
+const configs = browsers.map(browser => {
+  return helpers: { 
+    WebDriver: { browser }
+  }
+});
 
-```js
-multiple: {
-  // 2x chunks + 2x browsers = 4
-  parallel: {
-    // Splits tests into chunks
-    chunks: 2,
-    // run all tests in chrome and firefox
-    browsers: ["chrome", "firefox"]
-  },
+for (const config of configs) {
+  for (group of groupOfTests) {
+    const worker = workers.spawn();
+    worker.addTests(group);
+    worker.addConfig(config);
+  }
 }
-```
 
-Passing a function will enable you to provide your own chunking algorithm. The first argument passed to you function is an array of all test files, if you enabled grep the test files passed are already filtered to match the grep pattern.
+workers.run();
 
-```js
-multiple: {
-  parallel: {
-    // Splits tests into chunks by passing an anonymous function,
-    // only execute first and last found test file
-    chunks: (files) => {
-      return [
-        [ files[0] ], // chunk 1
-        [ files[files.length-1] ], // chunk 2
-      ]
-    },
-    // run all tests in chrome and firefox
-    browsers: ["chrome", "firefox"]
-  }
-}
-```
+workers.on(event.all.result, (status, completed, workerStats) => {
```

Comment:
@koushikmohan1996 I have really no idea what to write here in the docs
Can you help me? What are those objects and how to use them? :) 
Maybe we can simplify API to it would be bit more clear?

Reply:
Ok I will write the docs